### PR TITLE
[Locale] Removed enable/disable functionality

### LINKED ISSUE #7309: [RFC] Non Toggleable Locales
It seems there's a problem with the `Locale`s resolving that no one saw coming,
and now it bites our butts.

Issue:
=====
If you set a disabled locale as default in a channel, you'll break the whole shop.
Doesn't matter which locale is the customer using atm.

Possible solutions:
==============
Another resolver
-----------------
Which sole purpose of existence will be to fallback to the locale configured in config.yml

Pros:
- The site won't blow up

Cons:
 - You might not know the users are seeing everything in chinese

Restricting Locales used by channels
-------------------------------------
Blocking toggleability of default locales, and removing disabled locales from channel's edit select

Pros:
 - Site will stay up (probably)

Cons:
- Complexity goes up (esp. in forms, which we all love wholeheartedly);
- It will probably lead to way more bugs than it'll resolve;
- Crappy UX

Removing Locales' toggleability altogether
-------------------------------------------
Deleting the `enabled` field and all logic using it.

Pros:
- No bugs with disabling (obviously)
- Somewhat simplifies things

Cons:
- Another thing goes in the gutter (it would be a BC break, and will probably annoy some people heavily depending on it)

Personally I'm for the last one (hence the title), as it provides simplicity, and doesn't really have a downside since You'll still be able to "disable" a locale, by simply not adding it to the channel.

Either way those are my thoughts on the matter, and if anyone want to interject, provide more ups/downs or suggest another possible solution, please do so.

Cheers

### PR DESCRIPTION
**Pull request by NeverResponse**
| Q               | A
| --------------- | ---
| Bug fix?        | more of a precaution |
| New feature?    | no |
| BC breaks?      | yes |
| Related tickets | explained in, closes #7309 |
| License         | MIT |

### CODE REVIEW

File: tests/DataFixtures/ORM/resources/locales.yml [hunk 1]
```diff
@@ -1,7 +1,5 @@
 Sylius\Component\Locale\Model\Locale:
     locale_{en, nl, fr}:
         code: <current()>
-        enabled: true
     locale_de:
         code: de
-        enabled: false
```
Comment (michalmarcinkowski) [inline tests/DataFixtures/ORM/resources/locales.yml]: No need to have `de` locale defined separately anymore.

File: src/Sylius/Bundle/CoreBundle/Form/Extension/LocaleTypeExtension.php [hunk 2]
```diff
@@ -59,23 +52,13 @@ public function buildForm(FormBuilderInterface $builder, array $options)
 
             $locale = $event->getData();
             if ($locale instanceof LocaleInterface && null !== $locale->getCode()) {
-                $options['disabled'] = true;
```
Comment (pamil) [inline src/Sylius/Bundle/CoreBundle/Form/Extension/LocaleTypeExtension.php]: I guess it shouldn't be removed.

Comment (NeverResponse) [inline src/Sylius/Bundle/CoreBundle/Form/Extension/LocaleTypeExtension.php]: Went a bit overboard on this one. 👌

**Comment by michalmarcinkowski** [comment]
Thanks Janek! 👍

**Pull request merged**